homeassistant/components/device_tracker/luci.py

```diff
@@ -31,151 +34,69 @@
 
+        if filter_named:
+            return filter_named[0]
+        return None
+
+    @Throttle(MIN_TIME_BETWEEN_SCANS)
     def _update_info(self):
-        """Ensure the information from the Luci router is up to date.
+        """Check the Luci router for devices
 
-        Returns boolean if scanning successful.
+        Returns a boolean. True if scan was successful.
         """
-        if not self.success_init:
-            return False
-
-        _LOGGER.info("Checking ARP")
-
-        url = '{}/cgi-bin/luci/rpc/sys'.format(self.origin)
-
-        try:
-            result = _req_json_rpc(
-                url, 'net.arptable', params={'auth': self.token})
-        except InvalidLuciTokenError:
-            _LOGGER.info("Refreshing token")
-            self.refresh_token()
-            return False
-
-        if result:
-            self.last_results = []
-            for device_entry in result:
-                # Check if the Flags for each device contain
-                # NUD_REACHABLE and if so, add it to last_results
-                if int(device_entry['Flags'], 16) & 0x2:
-                    self.last_results.append(Device(device_entry['HW address'],
-                                                    device_entry['IP address'],
-                                                    device_entry['Flags'],
-                                                    device_entry['Device'],
-                                                    self.host))
-
-            return True
-
-        return False
-
-
-def _req_json_rpc(url, method, *args, **kwargs):
-    """Perform one JSON RPC operation."""
-    data = json.dumps({'method': method, 'params': args})
-
-    try:
-        res = requests.post(url, data=data, timeout=5, **kwargs)
-    except requests.exceptions.Timeout:
-        _LOGGER.exception("Connection to the router timed out")
-        return
-    if res.status_code == 200:
-        try:
-            result = res.json()
-        except ValueError:
-            # If json decoder could not parse the response
-            _LOGGER.exception("Failed to parse response from luci")
-            return
-        try:
-            return result['result']
-        except KeyError:
-            _LOGGER.exception("No result in response from luci")
-            return
-    elif res.status_code == 401:
-        # Authentication error
-        _LOGGER.exception(
-            "Failed to authenticate, check your username and password")
-        return
-    elif res.status_code == 403:
-        _LOGGER.error("Luci responded with a 403 Invalid token")
-        raise InvalidLuciTokenError
-
-    else:
-        _LOGGER.error("Invalid response from luci: %s", res)
-
-
-def _get_token(origin, username, password):
-    """Get authentication token for the given configuration."""
-    url = '{}/cgi-bin/luci/rpc/auth'.format(origin)
-    return _req_json_rpc(url, 'login', username, password)
+        _LOGGER.info("Scanning...")
```

Comment:
Replace this with a debug log that logs the result of `self.router.get_all_connected_devices`.